lib/nerves_hub_web/live/firmware.ex

```diff
@@ -151,6 +162,20 @@ defmodule NervesHubWeb.Live.Firmware do
     end
   end
 
+  defp error_feedback(
+         socket,
+         %Ecto.Changeset{errors: [uuid: {"has already been taken", _}]} = changeset
+       ) do
+    socket
+    |> assign(:error_message, "Duplicate Firmware Is Not Allowed")
+    |> assign(:error_code, :duplicate_firmware)
+    |> assign(
+      :firmware_path,
+      "/org/#{socket.assigns.org.name}/#{socket.assigns.product.name}/firmware/#{changeset.changes.uuid}"
+    )
```

Comment:
```suggestion
    |> assign(:error_code, {:duplicate_firmware, uuid})
```

Since we have control of this structure, we could just include the duplicate UUID and let the case handling of that view build the URL for it (or whatever else). That way we're not so coupled to 2 independent pieces of data having to exist in the assigns (`@error_code` and `@firmware_path`)